src/main/webapp/mapScript.js

```diff
@@ -132,32 +87,120 @@ function addNewTableItem(name, id) {
-      } else {
-        console.log("Unsuccessful deletion");
-      }
-    });
+  // Show popup.
+  // setting.style.visibility = "visible";
+  settings.classList.toggle("show");
+  document.getElementById("marker-name").value = markersArray[contentId].name;
+  document.getElementById("visit-hour").value =
+    markersArray[contentId].visitHour;
+  document.getElementById("visit-minute").value =
+    markersArray[contentId].visitMinute;
+  document.getElementById("leave-hour").value =
+    markersArray[contentId].leaveHour;
+  document.getElementById("leave-minute").value =
+    markersArray[contentId].leaveMinute;
+}
+
+function updateMarkerSettings(contentId) {
+  let markerName = document.getElementById("marker-name").value,
+    visitHour = document.getElementById("visit-hour").value,
+    visitMinute = document.getElementById("visit-minute").value,
+    leaveHour = document.getElementById("leave-hour").value,
+    leaveMinute = document.getElementById("leave-minute").value;
+
+  markersArray[contentId].name = markerName;
+  markersArray[contentId].visitHour = visitHour;
+  markersArray[contentId].visitMinute = visitMinute;
+  markersArray[contentId].leaveHour = leaveHour;
+  markersArray[contentId].leaveMinute = leaveMinute;
+
+  let tableItemElemens = document.getElementById("place" + contentId)
+    .childNodes;
+  tableItemElemens[1].innerHTML = markerName;
+}
+
+async function createRoute() {
+  var routeName = document.getElementById("route-name").value;
+  if (routeName == "") {
+    alert("Please add a name to your new route!");
+  } else {
+    var markersIds = [];
+    console.log(markersArray);
+
+    for (var i = 0; i < markersArray.length; i++) {
+      let lat = markersArray[i].marker.position.lat(),
+        lng = markersArray[i].marker.position.lng();
+      (markerName = markersArray[i].name),
+        (visitHour = markersArray[i].visitHour),
+        (visitMinute = markersArray[i].visitMinute),
+        (leaveHour = markersArray[i].leaveHour),
+        (leaveMinute = markersArray[i].leaveMinute);
+
+      let markerData = {
+        lat,
+        lng,
+        visitHour,
+        visitMinute,
+        leaveHour,
+        leaveMinute,
+        markerName,
+      };
+
+      let options = {
+        method: "POST",
+        body: JSON.stringify(markerData),
+        headers: {
+          "Content-Type": "application/json",
+        },
+      };
+
+      await fetch("/markers", options)
+        .then((response) => response.json())
+        .then((id) => {
+          console.log("receive new place's id " + id.toString());
+          markersIds.push({ id: id });
+        });
+    }
+
+    let options = {
+      method: "POST",
+      body: JSON.stringify(markersIds),
+      headers: {
+        "Content-Type": "application/json",
+      },
+    };
+
+    let URL = "/storeRoute?routeName=" + routeName;
+    await fetch(URL, options).then();
+
+    document.getElementById("places-table").innerHTML = "";
```

Comment:
With that line you as well delete "settings section".

Reply:
Solved, thanks!